midp/gfx.js

```diff
@@ -112,8 +112,34 @@ var currentlyFocusedTextEditor;
         return 1;
     };
 
+    var hasNewFrame = true;
+    var ctxToExecuteOnNewFrame;
+    function gotNewFrame(timestamp) {
+        if (ctxToExecuteOnNewFrame) {
+            ctxToExecuteOnNewFrame.execute();
+            ctxToExecuteOnNewFrame = null;
```

Comment:
I'm not sure why but I had imagined that `Context.execute` ran asynchronously (i.e. calling it would return immediately and then on the next step of the event loop the VM would actually start up again). Looking at its definition, that's definitely not the case, so the issue you point out is a real one. I'll fix this and update the PR